src/shared/utils/vis.ts

```diff
@@ -456,3 +461,115 @@ export const getDetectCoordinatingFields = (table: Table) => {
   }
   return true
 }
+
+const latLonAsTags = latLonColumns => {
+  if (latLonColumns?.lat?.key === 'tag' && latLonColumns?.lon?.key === 'tag') {
+    return true
+  }
+  return false
+}
+
+const getCoordinateColumnFlagged = (
+  table: Table,
+  useS2CellID: boolean,
+  s2Column: string,
+  latLonColumns: LatLonColumns
+): string => {
+  try {
+    if (useS2CellID && s2Column !== '') {
+      const column = table.getColumn(s2Column || 's2_cell_id')
+      if (column != null) {
+        return CoordinateType.S2
+      }
+    }
+    if (latLonAsTags(latLonColumns)) {
+      const lat = table.getColumn(latLonColumns?.lat?.column || 'lat')
+      const lon = table.getColumn(latLonColumns?.lon?.column || 'lon')
+
+      if (lat !== null && lon !== null) {
+        return CoordinateType.Tags
+      }
+    }
+
+    const latCoordinate = getColumnValue(
+      table,
+      latLonColumns?.lat?.column || 'lat'
+    )
+    const lonCoordinate = getColumnValue(
+      table,
+      latLonColumns?.lon?.column || 'lon'
+    )
+
+    if (latCoordinate && lonCoordinate) {
+      return CoordinateType.Fields
+    }
+
+    return CoordinateType.None
+  } catch (e) {
+    throw new Error('lat_lon_not_found')
+  }
+}
+
+export const getGeoCoordinatesFlagged = (
+  table: Table,
+  index: number,
+  useS2CellID: boolean,
+  s2Column: string,
+  latLonColumns: LatLonColumns
+): {lon: number; lat: number} | null => {
+  const coordinateColumn = getCoordinateColumnFlagged(
+    table,
+    useS2CellID,
+    s2Column,
+    latLonColumns
+  )
+
+  switch (coordinateColumn) {
+    case CoordinateType.S2:
+      return getCoordinateFromS2(table, index)
+    case CoordinateType.Tags:
+      const latColumn = table.getColumn(latLonColumns?.lat?.column || 'lat')
+      const lonColumn = table.getColumn(latLonColumns?.lon?.column || 'lon')
+      return {
+        lat: parseCoordinates(latColumn[index]),
+        lon: parseCoordinates(lonColumn[index]),
+      }
+    case CoordinateType.Fields:
+      const latCoordinate = getColumnValue(
+        table,
+        latLonColumns?.lat?.column || 'lat'
+      )
+      const lonCoordinate = getColumnValue(
+        table,
+        latLonColumns?.lon?.column || 'lon'
+      )
+      return {
+        lat: parseCoordinates(latCoordinate),
+        lon: parseCoordinates(lonCoordinate),
+      }
+    default:
+      throw new Error('lat_lon_not_provided')
+  }
+}
+
+export const getDetectCoordinatingFieldsFlagged = (
+  table: Table,
+  useS2CellID: boolean,
+  s2Column: string,
+  latLonColumns: LatLonColumns
+) => {
+  const coordinateColumn = getCoordinateColumnFlagged(
+    table,
+    useS2CellID,
+    s2Column,
+    latLonColumns
+  )
+
+  if (
+    coordinateColumn === CoordinateType.S2 ||
+    coordinateColumn === CoordinateType.Tags
+  ) {
+    return false
+  }
+  return true
+}
```

Comment:
question, what is the `flagged` part of the methods? I was trying to see where they reference feature flags but couldn't find it